pkg/kubelet/kubelet.go

```diff
@@ -775,6 +780,14 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 
 	tokenManager := token.NewManager(kubeDeps.KubeClient)
 
+	// Initialize csiDriverLister before calling NewInitializedVolumePluginMgr
+	// because csiDriverLister is needed in csi_plugin.go
+	if utilfeature.DefaultFeatureGate.Enabled(features.CSIDriverRegistry) {
+		klet.informerFactory = informers.NewSharedInformerFactory(klet.kubeClient, resyncPeriod)
+		csiDriverInformer := klet.informerFactory.Storage().V1beta1().CSIDrivers()
+		klet.csiDriverLister = csiDriverInformer.Lister()
```

Comment:
Check for save `HasSynced` variable into kubelet to wait for in `Run()`?

Reply:
WaitForCacheSync is called in volumeManager Run().  That should be okay?